Public/Src/Engine/Processes/Tracing/Log.cs

```diff
@@ -1020,14 +1021,15 @@ public abstract partial class Logger
             (ushort)LogEventId.LogPhaseDuration,
             EventGenerators = EventGenerators.LocalOnly,
             EventLevel = Level.Verbose,
-            Keywords = (int)Keywords.UserMessage,
-            EventTask = (ushort)Tasks.Engine,
-            Message = "{pipDescription} -- Done with phase '{phaseName}' in {duration}.  {extraInfo}")]
-        public abstract void LogPhaseDuration(LoggingContext context, string pipDescription, string phaseName, string duration, string extraInfo);
+            Keywords = (int)Keywords.Diagnostics,
+            EventTask = (ushort)Tasks.PipExecutor,
+            Message = "[{pipSemiStableHash}] Done with phase '{phaseName}' in {duration}.  {extraInfo}")]
+        public abstract void LogPhaseDuration(LoggingContext context, string pipSemiStableHash, string phaseName, string duration, string extraInfo);
 
-        public void LogSubPhaseDuration(LoggingContext context, string pipDescription, string phaseName, TimeSpan duration, string extraInfo = "")
+        public void LogSubPhaseDuration(LoggingContext context, Pip pip, SandboxedProcessFactory.SandboxedProcessCounters counter, TimeSpan duration, string extraInfo = "")
         {
-            LogPhaseDuration(context, pipDescription, phaseName, duration.ToString(), extraInfo);
+            SandboxedProcessFactory.Counters.AddToCounter(counter, duration);
+            LogPhaseDuration(context, pip.FormattedSemiStableHash, counter.ToString(), duration.ToString(), extraInfo);

```

Comment:
>counter.ToString() [](start = 67, length = 18)

This used to have bad performance too under FullFramework. Luckily NetCore removed the hidden reflection :)